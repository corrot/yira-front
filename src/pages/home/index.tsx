import React from 'react';
import { useTranslation } from 'react-i18next';

import Api from '@/services/api';
import { Flex } from '@/styled/flex';
import useApi from '@/hooks/useApi';

export const Home = () => {
	const { t } = useTranslation();
	const [users] = useApi(Api.user.getUsers);

	const handleLogin = () => {
		try {
			const data = Api.user.authenticateUser();
			console.log(data);
		} catch (err) {
			console.log(err);
		}
	};

	const handleRegistration = () => {
		try {
			const data = Api.user.registerUser();
			console.log(data);
		} catch (err) {
			console.log(err);
		}
	};

	return (
		<>
			<Flex center>
				<Flex center sm={6} xs={8} padding="1rem">
					<span>{t('Main page')}</span>
				</Flex>
				<Flex direction="column" center sm={6} xs={8} padding="1rem">
					{users.map(u => (
						<div key={u.id}>{u.username}</div>
					))}
				</Flex>
			</Flex>
			<button type="button" onClick={handleLogin}>
				login
			</button>
			<button type="button" onClick={handleRegistration}>
				register
			</button>
		</>
	);
};

export default Home;
