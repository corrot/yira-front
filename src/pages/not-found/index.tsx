import React, { FC, ReactElement } from 'react';

export const NotFound: FC = (): ReactElement<HTMLDivElement> => (
	<div className="c-not-found">
		404 <br />
		Page not found
	</div>
);

export default NotFound;
