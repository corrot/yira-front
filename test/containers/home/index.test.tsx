import React from 'react';
import renderer from 'react-test-renderer';

import { Home } from '@/pages/home';

describe('Home component', () => {
	it('should render successfully', () => {
		const tree = renderer.create(<Home />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
