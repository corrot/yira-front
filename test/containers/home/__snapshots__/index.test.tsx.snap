// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render successfully 1`] = `
<div
  className="sc-bdVaJa fDpJja"
>
  <div
    className="sc-bdVaJa PqXEI"
  >
    <span>
      HOME
    </span>
  </div>
</div>
`;
